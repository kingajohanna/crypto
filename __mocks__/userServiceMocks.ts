export const adduser = {
    status: "success",
    message: "user added successfully",
    data: {
        id: "DCzgasdfadKdDpkAWqHEw",
        email: "test@gmail.com",
        createdAt: 1648314794066,
        name: "bela",
        photoURL: "https://lh3.googleusercontent.com/a-/AOh14GjVLepb70SMpFXeAyqty8h58J9TQY7lw8wwsx4Oxg=s96-d",
    },
};

export const removeuser = {
    status: "success",
    message: "user removed successfully",
};

export const addfav = {
    status: "success",
    message: "coin added to favs successfully",
};

export const removefav = {
    status: "success",
    message: "coin removed from favs successfully",
};

export const addcoin = {
    status: "success",
    message: "coin added successfully",
    data: {
        id: "QjuLdgDdM1CkMnEMZOZg",
        userId: "DCzgasdfadKdDpkAWqH",
        coinId: "storj",
        holdings: 5,
        price: 2000,
        currency: "huf",
    },
};

export const getcoins = {
    status: "success",
    message: "fetch owned coins",
    data: [
        {
            userId: "DCzgasdfadKdDpkAWqH",
            id: "7s9J5XTfdaOthk0QIbMP",
            currency: "huf",
            coinId: "storj",
            holdings: 5,
            price: 2000,
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            userId: "DCzgasdfadKdDpkAWqH",
            currency: "huf",
            holdings: 5,
            price: 2000,
            id: "QjuLdgDdM1CkMnEMZOZg",
            coinId: "storj",
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            price: 159492,
            coinId: "storj",
            userId: "DCzgasdfadKdDpkAWqH",
            currency: "huf",
            id: "VB8cBlj3fRJ7cPSRiqFg",
            holdings: 9,
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            id: "ZlH7U70fK7ChTCTogZfz",
            holdings: 5,
            currency: "huf",
            price: 2000,
            userId: "DCzgasdfadKdDpkAWqH",
            coinId: "storj",
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            coinId: "storj",
            holdings: 5,
            price: 2000,
            id: "hssyU4lUp7R53eztlUsA",
            userId: "DCzgasdfadKdDpkAWqH",
            currency: "huf",
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
    ],
};

export const setcoin = {
    status: "success",
    message: "coin edited successfully",
};
