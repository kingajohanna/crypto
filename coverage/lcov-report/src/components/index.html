
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.06% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>232/305</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">31.25% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>30/96</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">27.27% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>18/66</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">77.57% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>211/272</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="AccountBottomModal.tsx"><a href="AccountBottomModal.tsx.html">AccountBottomModal.tsx</a></td>
	<td data-value="83.33" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 83%"></div><div class="cover-empty" style="width: 17%"></div></div>
	</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="81.81" class="pct high">81.81%</td>
	<td data-value="11" class="abs high">9/11</td>
	</tr>

<tr>
	<td class="file high" data-value="AccountButton.tsx"><a href="AccountButton.tsx.html">AccountButton.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="50" class="abs high">50/50</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="28" class="abs high">28/28</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="43" class="abs high">43/43</td>
	</tr>

<tr>
	<td class="file medium" data-value="Alert.tsx"><a href="Alert.tsx.html">Alert.tsx</a></td>
	<td data-value="54.16" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 54%"></div><div class="cover-empty" style="width: 46%"></div></div>
	</td>
	<td data-value="54.16" class="pct medium">54.16%</td>
	<td data-value="24" class="abs medium">13/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="47.61" class="pct low">47.61%</td>
	<td data-value="21" class="abs low">10/21</td>
	</tr>

<tr>
	<td class="file low" data-value="Autocomplete.tsx"><a href="Autocomplete.tsx.html">Autocomplete.tsx</a></td>
	<td data-value="34.37" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 34%"></div><div class="cover-empty" style="width: 66%"></div></div>
	</td>
	<td data-value="34.37" class="pct low">34.37%</td>
	<td data-value="32" class="abs low">11/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	</tr>

<tr>
	<td class="file high" data-value="Button.tsx"><a href="Button.tsx.html">Button.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="CryptoChart.tsx"><a href="CryptoChart.tsx.html">CryptoChart.tsx</a></td>
	<td data-value="73.07" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 73%"></div><div class="cover-empty" style="width: 27%"></div></div>
	</td>
	<td data-value="73.07" class="pct medium">73.07%</td>
	<td data-value="26" class="abs medium">19/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="24" class="abs medium">18/24</td>
	</tr>

<tr>
	<td class="file medium" data-value="CryptoCoin.tsx"><a href="CryptoCoin.tsx.html">CryptoCoin.tsx</a></td>
	<td data-value="72" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 72%"></div><div class="cover-empty" style="width: 28%"></div></div>
	</td>
	<td data-value="72" class="pct medium">72%</td>
	<td data-value="25" class="abs medium">18/25</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="78.26" class="pct medium">78.26%</td>
	<td data-value="23" class="abs medium">18/23</td>
	</tr>

<tr>
	<td class="file high" data-value="EditOwnedCoinBottomModal.tsx"><a href="EditOwnedCoinBottomModal.tsx.html">EditOwnedCoinBottomModal.tsx</a></td>
	<td data-value="84.21" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="19" class="abs high">16/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="18" class="abs high">15/18</td>
	</tr>

<tr>
	<td class="file high" data-value="ErrorComponent.tsx"><a href="ErrorComponent.tsx.html">ErrorComponent.tsx</a></td>
	<td data-value="87.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="InputField.tsx"><a href="InputField.tsx.html">InputField.tsx</a></td>
	<td data-value="69.23" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 69%"></div><div class="cover-empty" style="width: 31%"></div></div>
	</td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	</tr>

<tr>
	<td class="file high" data-value="NewOwnedCoinBottomModal.tsx"><a href="NewOwnedCoinBottomModal.tsx.html">NewOwnedCoinBottomModal.tsx</a></td>
	<td data-value="85" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 85%"></div><div class="cover-empty" style="width: 15%"></div></div>
	</td>
	<td data-value="85" class="pct high">85%</td>
	<td data-value="20" class="abs high">17/20</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="19" class="abs high">16/19</td>
	</tr>

<tr>
	<td class="file medium" data-value="OwnedCoins.tsx"><a href="OwnedCoins.tsx.html">OwnedCoins.tsx</a></td>
	<td data-value="75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 75%"></div><div class="cover-empty" style="width: 25%"></div></div>
	</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="24" class="abs medium">18/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="78.26" class="pct medium">78.26%</td>
	<td data-value="23" class="abs medium">18/23</td>
	</tr>

<tr>
	<td class="file medium" data-value="ScreenBackground.tsx"><a href="ScreenBackground.tsx.html">ScreenBackground.tsx</a></td>
	<td data-value="64.7" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 64%"></div><div class="cover-empty" style="width: 36%"></div></div>
	</td>
	<td data-value="64.7" class="pct medium">64.7%</td>
	<td data-value="17" class="abs medium">11/17</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="71.42" class="pct medium">71.42%</td>
	<td data-value="14" class="abs medium">10/14</td>
	</tr>

<tr>
	<td class="file high" data-value="SelectedCoinGraph.tsx"><a href="SelectedCoinGraph.tsx.html">SelectedCoinGraph.tsx</a></td>
	<td data-value="90" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="88.88" class="pct high">88.88%</td>
	<td data-value="9" class="abs high">8/9</td>
	</tr>

<tr>
	<td class="file high" data-value="TabHeader.tsx"><a href="TabHeader.tsx.html">TabHeader.tsx</a></td>
	<td data-value="87.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

<tr>
	<td class="file high" data-value="UserRequiredContent.tsx"><a href="UserRequiredContent.tsx.html">UserRequiredContent.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Mon Apr 18 2022 01:12:07 GMT+0200 (GMT+02:00)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    