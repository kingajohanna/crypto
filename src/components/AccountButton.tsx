import React from "react";
import { Image, PressableProps } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import styled from "styled-components/native";

import { Colors } from "@theme/Colors";

enum ButtonVariant {
    google = "google",
    apple = "apple",
    facebook = "facebook",
    email = "email",
    logout = "logout",
    delete = "delete",
}

type ButtonProps = {
    variant: ButtonVariant;
} & PressableProps;

const ButtonComponent: React.FC<ButtonProps> = ({ variant, onPress }) => {
    return (
        <ButtonContainer variant={variant} onPress={onPress}>
            <IconContainer>{socialIcon(variant)}</IconContainer>
            <Text variant={variant}>{socialText(variant)}</Text>
        </ButtonContainer>
    );
};

const socialText = (variant: ButtonVariant) => {
    switch (variant) {
        case ButtonVariant.google:
            return "Continue with Google";
        case ButtonVariant.apple:
            return "Continue with Apple";
        case ButtonVariant.facebook:
            return "Continue with facebook";
        case ButtonVariant.logout:
            return "Tap to logout";
        case ButtonVariant.delete:
            return "Tap to delete account";
        case ButtonVariant.email:
        default:
            return "Continue with email";
    }
};

const socialIcon = (variant: ButtonVariant) => {
    switch (variant) {
        case ButtonVariant.google:
            return <Image style={{ width: 24, height: 24 }} source={require("@assets/googleLogo.png")} />;
        case ButtonVariant.apple:
            return <Icon name="logo-apple" size={24} color={Colors.white} />;
        case ButtonVariant.facebook:
            return <Icon name="logo-facebook" size={24} color={Colors.white} />;
        case ButtonVariant.logout:
            return <Icon name="log-out" size={24} color={Colors.fluorescentBlue} />;
        case ButtonVariant.delete:
            return <Icon name="person-remove" size={24} color={Colors.fluorescentBlue} />;
        case ButtonVariant.email:
        default:
            return <Icon name="mail" size={24} color={Colors.fluorescentBlue} />;
    }
};

const backgroundColor = (variant: ButtonVariant) => {
    switch (variant) {
        case ButtonVariant.facebook:
            return Colors.facebookBlue;
        case ButtonVariant.google:
            return Colors.white;
        case ButtonVariant.apple:
            return Colors.black;
        case ButtonVariant.logout:
        case ButtonVariant.delete:
        case ButtonVariant.email:
        default:
            return Colors.gunmetal;
    }
};

const fontColor = (variant: ButtonVariant) => {
    switch (variant) {
        case ButtonVariant.google:
            return Colors.gray;
        case ButtonVariant.apple:
        case ButtonVariant.facebook:
            return Colors.white;
        case ButtonVariant.email:
        case ButtonVariant.logout:
        case ButtonVariant.delete:
        default:
            return Colors.fluorescentBlue;
    }
};

const IconContainer = styled.View({
    marginRight: 15,
    alignItems: "center",
});

const Text = styled.Text<ButtonProps>((props: ButtonProps) => ({
    fontSize: 18,
    lineHeight: "24px",
    color: fontColor(props.variant),
}));

const ButtonContainer = styled.Pressable<ButtonProps>((props: ButtonProps) => ({
    marginTop: 12,
    paddingHorizontal: 15,
    backgroundColor: backgroundColor(props.variant),
    width: 342,
    height: 54,
    borderRadius: 10,
    alignItems: "center",
    flexDirection: "row",
    borderColor: Colors.cadetBlue,
    borderWidth: props.variant !== ButtonVariant.facebook && props.variant !== ButtonVariant.google ? 2 : 0,
}));

const Google = styled(ButtonComponent).attrs(() => ({
    variant: ButtonVariant.google,
}))({});
const Apple = styled(ButtonComponent).attrs(() => ({
    variant: ButtonVariant.apple,
}))({});
const Facebook = styled(ButtonComponent).attrs(() => ({
    variant: ButtonVariant.facebook,
}))({});
const Email = styled(ButtonComponent).attrs(() => ({
    variant: ButtonVariant.email,
}))({});
const Logout = styled(ButtonComponent).attrs(() => ({
    variant: ButtonVariant.logout,
}))({});
const Delete = styled(ButtonComponent).attrs(() => ({
    variant: ButtonVariant.delete,
}))({});

export const AccountButton = {
    Google: Google,
    Apple: Apple,
    Facebook: Facebook,
    Email: Email,
    Logout: Logout,
    Delete: Delete,
};
