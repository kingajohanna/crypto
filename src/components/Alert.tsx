import React, { useState } from "react";
import Dialog from "react-native-dialog";
import { deleteAccount, logOut } from "@services/FirebaseServices";
import { Alert, View } from "react-native";

type AlertProps = {
    visible: boolean;
    setVisible: (arg0: boolean) => void;
};

/*
    alert with textbox what shows when the users try to to delete their account
*/
export const DeleteAlert: React.FC<AlertProps> = ({ visible, setVisible }) => {
    const [text, setText] = useState("");
    return (
        <View>
            <Dialog.Container visible={visible}>
                <Dialog.Title>Account delete</Dialog.Title>
                <Dialog.Description>Do you want to delete this account? Please enter 'DELETE' in the editbox.</Dialog.Description>
                <Dialog.Input onChangeText={(text) => setText(text)} value={text} />
                <Dialog.Button
                    label="Cancel"
                    onPress={() => {
                        setText("");
                        setVisible(false);
                    }}
                />
                <Dialog.Button
                    label="Delete"
                    onPress={() => {
                        if (text === "DELETE") {
                            deleteAccount();
                            setVisible(false);
                        }
                    }}
                />
            </Dialog.Container>
        </View>
    );
};

/*
    alert that shows when the user try to logout
*/
export const LogoutAlert = () => {
    Alert.alert("Logout", `Click OK to logout!`, [
        {
            text: "Cancel",
            onPress: () => {},
        },
        {
            text: "OK",
            onPress: () => logOut(),
        },
    ]);
};

/*
    to delete google sso in account if the user logged in earlier then 5 minutes
    #errorhandling
*/
export const ReLogin = () => {
    Alert.alert(
        "",
        `This operation is sensitive and requires recent authentication. 
Log in again before retrying this request.`,
        [
            {
                text: "OK",
                onPress: () => logOut(),
            },
        ],
    );
};

export const PasswordReset = () => {
    Alert.alert("", `The password reset link has been sent to your email address`, [
        {
            text: "OK",
            onPress: () => {},
        },
    ]);
};
