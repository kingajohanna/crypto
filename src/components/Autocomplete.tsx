import React, { useState } from "react";
import { TextInputProps, ScrollView } from "react-native";
import { Menu } from "react-native-paper";
import styled from "styled-components/native";
import { Colors, hexToRGBA } from "@theme/Colors";
import { KeyValue } from "@constants/DataTypes";

type AutocompleteProps = {
    width?: number;
    data: KeyValue[];
} & TextInputProps;

/*
    textinput with scrollable suggestion
*/
export const Autocomplete: React.FC<AutocompleteProps> = ({ data, placeholder, onChangeText, width }) => {
    const [value, setValue] = useState("");
    const [menuVisible, setMenuVisible] = useState(false);
    const [filteredData, setFilteredData] = useState<KeyValue[]>(data);

    const filterData = (text: string) => {
        return data.filter((item) => item.name.toLowerCase().startsWith(text.toLowerCase()));
    };

    return (
        <Container>
            <StyledInput
                width={width}
                placeholderTextColor={hexToRGBA(Colors.cadetBlue, 0.5)}
                placeholder={placeholder}
                value={value}
                onChangeText={(text) => {
                    if (text.length > 0) {
                        setFilteredData(filterData(text));
                    } else if (text.length === 0) {
                        setFilteredData([]);
                    }
                    setMenuVisible(true);
                    setValue(text);
                }}
                onFocus={() => {
                    setMenuVisible(true);
                }}
                onBlur={() => setMenuVisible(false)}
            />

            {menuVisible && filteredData && (
                <SearchContainer>
                    <ScrollView style={{ position: "absolute", borderColor: Colors.cadetBlue, borderWidth: 1, borderRadius: 5, maxHeight: 230, backgroundColor: Colors.gunmetal }}>
                        {filteredData.map((word, i) => (
                            <Menu.Item
                                key={i}
                                onPress={() => {
                                    onChangeText!(word.id);
                                    setValue(word.name);
                                    setMenuVisible(false);
                                }}
                                style={{ backgroundColor: Colors.gunmetal, height: 40, width: 150 }}
                                titleStyle={{ color: Colors.fluorescentBlue }}
                                title={word.name}
                            />
                        ))}
                    </ScrollView>
                </SearchContainer>
            )}
        </Container>
    );
};

const Container = styled.View({
    alignItems: "flex-start",
    padding: 8,
});

const SearchContainer = styled.View({
    flexDirection: "column",
    width: 150,
    height: 16,
    backgroundColor: Colors.gunmetal,
});

export const StyledInput = styled.TextInput<Partial<AutocompleteProps>>(({ width }) => ({
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: Colors.gunmetal,
    height: 40,
    width: width ? width : 150,
    borderWidth: 1,
    borderColor: Colors.fluorescentBlue,
    borderRadius: 4,
    color: Colors.cadetBlue,
    paddingHorizontal: 16,
}));
