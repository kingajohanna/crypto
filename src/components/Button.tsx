import React from "react";
import { PressableProps } from "react-native";
import styled from "styled-components/native";
import { Colors } from "@theme/Colors";

type ButtonProps = {
    title: string;
} & PressableProps;

/*
    basic button
*/
export const Button: React.FC<ButtonProps> = ({ title, onPress }) => {
    return (
        <Container onPress={onPress}>
            <Text>{title}</Text>
        </Container>
    );
};

//export for jest test
export const Container = styled.Pressable({
    backgroundColor: Colors.fluorescentBlue,
    width: 120,
    height: 44,
    borderWidth: 2,
    borderRadius: 15,
    alignItems: "center",
    justifyContent: "center",
    borderColor: Colors.cadetBlue,
});

const Text = styled.Text({
    color: Colors.gunmetal,
    fontSize: 18,
});
