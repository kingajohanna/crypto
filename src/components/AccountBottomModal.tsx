import React from "react";
import styled from "styled-components/native";
import { Button } from "@components/Button";
import { ErrorText } from "@components/ErrorText";

type ModalProps = {
    errorText?: string;
    primaryButtonText: string;
    secondaryButtonText: string;
    primaryButtonOnPress: Function;
    secondaryButtonOnPress: Function;
};

/*
    bottom modal, with optional error text for the Account screen
*/
export const AccountBottomModal: React.FC<ModalProps> = ({ errorText, children, primaryButtonText, secondaryButtonText, primaryButtonOnPress, secondaryButtonOnPress }) => {
    return (
        <Container>
            {!!errorText && <ErrorText>{errorText}</ErrorText>}
            <ChildContainer>{children}</ChildContainer>
            <ButtonContainer>
                <Button title={secondaryButtonText} onPress={() => secondaryButtonOnPress()} />
                <Button title={primaryButtonText} onPress={() => primaryButtonOnPress()} />
            </ButtonContainer>
        </Container>
    );
};

const Container = styled.View({
    paddingTop: 24,
});

const ChildContainer = styled.View({
    alignItems: "center",
    paddingBottom: 16,
});

const ButtonContainer = styled.View({
    flexDirection: "row",
    justifyContent: "space-between",
    paddingHorizontal: 24,
});
