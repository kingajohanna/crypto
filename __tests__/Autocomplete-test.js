import React from "react";
import { Autocomplete } from "../src/components/Autocomplete";
import "jest-styled-components";
import ShallowRenderer from "react-test-renderer/shallow";
import { coins } from "../__mocks__/cryptoServiceMocks";

test("Onpress called correctly", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Autocomplete placeholder="Coins" onChangeText={jest.fn()} data={coins} />);
    expect(result).toMatchSnapshot();
});
