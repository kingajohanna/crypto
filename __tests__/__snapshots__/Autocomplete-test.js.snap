// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onpress called correctly 1`] = `
<Styled(View)>
  <Styled(TextInput)
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Coins"
    placeholderTextColor="rgba(69, 162, 158, 0.5)"
    value=""
  />
</Styled(View)>
`;
